src/DGtal/geometry/curves/AlphaThickSegmentComputer.ih

```diff
@@ -0,0 +1,965 @@
+  out << std::endl << "Segment thickness : " << myState.actualThickness << std::endl;
+  PointD pt1, pt2, pt3, pt4;
+  getBoundingBox(pt1, pt2, pt3, pt4);
+  out << std::endl << "Bounding box:" << pt1 << " " << pt2 << " " << pt3 << " " << pt4 << std::endl;
+
+}
+
+
+
+
+//-----------------------------------------------------------------------------
+template < typename TInputPoint, typename TConstIterator>
+inline
+bool
+DGtal::AlphaThickSegmentComputer< TInputPoint, TConstIterator>::isValid()  const
+{
+  return true;
+}
+
+//-----------------------------------------------------------------------------
+template < typename TInputPoint, typename TConstIterator>
+inline
+std::pair<std::pair<TInputPoint, TInputPoint>, TInputPoint>
+DGtal::AlphaThickSegmentComputer< TInputPoint, TConstIterator>::getAntipodalLeaningPoints() const
+  {
+    std::pair<std::pair<TInputPoint, TInputPoint>, TInputPoint> r;
+    std::pair<TInputPoint, TInputPoint> edgePair;
+    edgePair.first = myState.edgePh;
+    edgePair.second = myState.edgeQh;
+    r.first = edgePair;
+    r.second = myState.vertexSh;
+    return r;
+  }
+
+//-----------------------------------------------------------------------------
+template < typename TInputPoint, typename TConstIterator>
+inline
+std::pair<TInputPoint, TInputPoint >
+DGtal::AlphaThickSegmentComputer< TInputPoint, TConstIterator>::getExtremityPoints()  const
+{
+  std::pair<TInputPoint, TInputPoint> r;
+  r.first = myState.lastBack;
+  r.second = myState.lastFront;
+  computeExtremaPoints(myState.melkmanQueue.begin(), myState.melkmanQueue.end() , r.first, r.second);
+  return r;
+}
+
+
+
+
+
+
+
+//-----------------------------------------------------------------------------
+template < typename TInputPoint, typename TConstIterator>
+inline
+bool
+DGtal::AlphaThickSegmentComputer< TInputPoint, TConstIterator>::melkmanIsWellInitialized()  const
+{
+  return myState.isMelkmanInitialized;
+}
+
+
+//-----------------------------------------------------------------------------
+template < typename TInputPoint, typename TConstIterator>
+inline
+double
+DGtal::AlphaThickSegmentComputer< TInputPoint, TConstIterator>::melkmanUpdateMainHeight()  
+{
+  int queueSize = myState.melkmanQueue.size();
+  InputPoint P, Q, R, S;
+  int q,s; /* [pq] is an edge and s a vertex */
+  bool stop = false; /* Default is false */
+  double tmpheight, k, dMinus, dPlus;
+  double maxDoubleValue =  std::numeric_limits<int>::max();
+
+  /* Take the first edge and find the corresponding point of support */
+  P =  myState.melkmanQueue[ 0 ] ;
+  q = 1;
+  Q = myState.melkmanQueue[ q++ ] ;
+  /** To find s, start at q+1 and move until angle between the edge at s 
+   *  and the line pq is positive.
+   */
+  s = q;
+  S = myState.melkmanQueue[ s++ ];
+  R = myState.melkmanQueue[ s++ ];
+  while( !stop )
+    {
+      /* Use vectorial product to ckeck angle */
+      stop = ( ( R[0] - S[0] ) * ( Q[1] - P[1] ) - ( R[1] - S[1] ) * ( Q[0] - P[0] ) ) >= 0;
```

Comment:
Perhaps use the cross product implemented in class PointVector ? It is more a kind of Orientation primitive here.